compiler/rustc_mir_build/src/thir/pattern/deconstruct_pat.rs

```diff
@@ -1557,6 +1557,10 @@ impl<'p, 'tcx> DeconstructedPat<'p, 'tcx> {
                 let pats = expand_or_pat(pat);
                 fields = Fields::from_iter(cx, pats.into_iter().map(mkpat));
             }
+            PatKind::Never => {
+                ctor = Wildcard; // FIXME(never_patterns)
```

Comment:
While you're at it, can you elaborate this FIXME? Is this why `Some(!)` acts like `Some(_)`?

Reply:
Yeah, the exhaustiveness part of never patterns is the trickiest (and most exciting) bit. That's just a default value to make rustc compile. I'll edit the comment